src/error_log.cpp

```diff
@@ -81,5 +81,189 @@ void FFDCFile::removeFFDCFile()
     std::remove(_fileName.data());
 }
 
+FFDCFileSet::FFDCFileSet(const bool collectTraces, const json& calloutData)
+{
+    if (collectTraces)
+    {
+        try
+        {
+            createFFDCFilesForTraces();
+        }
+        catch (const std::exception& e)
+        {
+            lg2::error("Exception while collecting traces for FFDC: {ERROR}",
+                       "ERROR", e.what());
+        }
+    }
+
+    if (!calloutData.is_null())
+    {
+        try
+        {
+            createFFDCFilesForCallouts(calloutData);
+        }
+        catch (const std::exception& e)
+        {
+            lg2::error(
+                "Exception while collecting callout data for FFDC: {ERROR}",
+                "ERROR", e.what());
+        }
+    }
+}
+
+std::optional<std::string>
+    FFDCFileSet::getTraceFieldValue(sd_journal* journal,
+                                    const std::string& fieldName)
+{
+    const void* raw_data{nullptr};
+    size_t length{0};
+
+    auto rc = sd_journal_get_data(journal, fieldName.c_str(), &raw_data,
+                                  &length);
+    if (0 == rc)
+    {
+        std::string fieldValue(static_cast<const char*>(raw_data), length);
+
+        // The data returned by sd_journal_get_data will be prefixed with the
+        // field name and "=".
+        if (auto fieldValPos = fieldValue.find('=');
+            fieldValPos != std::string::npos)
+        {
+            // Just return field value
+            return fieldValue.substr(fieldValPos + 1);
+        }
+        else
+        {
+            // Should not happen as per sd_journal_get_data documentation
+            lg2::error(
+                "Failed to find the journal field separator [=] in the retrieved field {FIELD_VALUE}",
```

Comment:
Did you ran the local CI? its crossing 80 line character per line. The comment applicable into all places.

Please follow guideline.